src/Rules/PHPUnit/AttributeRequiresPhpVersionRule.php

```diff
@@ -31,12 +37,16 @@ class AttributeRequiresPhpVersionRule implements Rule
 	public function __construct(
 		PHPUnitVersion $PHPUnitVersion,
 		TestMethodsHelper $testMethodsHelper,
-		bool $deprecationRulesInstalled
+		bool $deprecationRulesInstalled,
+		PhpVersion $phpVersion
 	)
 	{
 		$this->PHPUnitVersion = $PHPUnitVersion;
 		$this->testMethodsHelper = $testMethodsHelper;
 		$this->deprecationRulesInstalled = $deprecationRulesInstalled;
+
+		$parser = new VersionParser();
+		$this->phpstanVersionConstraint = $parser->parseConstraints($phpVersion->getVersionString());
```

Comment:
maybe this should compare against the composer.json min version instead of the phpstan.neon version?